packages/documentation/src/lib/playground.tsx

```diff
@@ -0,0 +1,344 @@
+      },
+    ],
+    "typescript-koa": [
+      {
+        uri: "https://unpkg.com/@nahkies/typescript-koa-runtime@latest/package.json",
+        path: "/node_modules/@nahkies/typescript-koa-runtime/package.json",
+      },
+      {
+        uri: "https://unpkg.com/@nahkies/typescript-koa-runtime@latest/dist/server.d.ts",
+        path: "/node_modules/@nahkies/typescript-koa-runtime/server.d.ts",
+      },
+      {
+        uri: "https://unpkg.com/@nahkies/typescript-koa-runtime@latest/dist/errors.d.ts",
+        path: "/node_modules/@nahkies/typescript-koa-runtime/errors.d.ts",
+      },
+      {
+        uri: "https://unpkg.com/@nahkies/typescript-koa-runtime@latest/dist/zod.d.ts",
+        path: "/node_modules/@nahkies/typescript-koa-runtime/zod.d.ts",
+      },
+      {
+        uri: "https://unpkg.com/@nahkies/typescript-koa-runtime@latest/dist/joi.d.ts",
+        path: "/node_modules/@nahkies/typescript-koa-runtime/joi.d.ts",
+      },
+    ],
+  }
+
+  for (const file of files[template]) {
+    const source = await (await fetch(file.uri)).text()
+    const uri = monaco.Uri.file(file.path)
+    if (!monaco.editor.getModel(uri)) {
+      console.info("[MICHAEL] createModel for " + uri.path)
+      monaco.editor.createModel(source, "typescript", uri)
+      // monaco.languages.typescript.typescriptDefaults.addExtraLib(await source.text(), file.path)
+    }
+  }
+}
+
+const Playground: React.FC<{
+  specifications: {filename: string; content: string}[]
+}> = ({specifications}) => {
+  const {
+    formatter,
+    openapiValidator,
+    typespecLoader,
+    host,
+    selectedModel,
+    availableModels,
+    setSelectedModel,
+  } = useHost({specifications})
+  const [input, setInput] = useState<string>()
+  const [monaco, setMonaco] = useState<Monaco>()
+
+  const [inputEditor, setInputEditor] = useState<editor.IStandaloneCodeEditor>()
+  const [outputEditor, setOutputEditor] =
+    useState<editor.IStandaloneCodeEditor>()
+
+  const onMountInput = useCallback((editor: editor.IStandaloneCodeEditor) => {
+    setInputEditor(editor)
+  }, [])
+
+  const inputModel = monaco?.editor.getModel(monaco.Uri.file("/example.yaml"))
+
+  useEffect(() => {
+    if (!inputModel || !inputEditor) {
+      return
+    }
+    console.info("mounting input", {inputModel})
+    inputEditor.setModel(inputModel)
+    inputModel.onDidChangeContent((e) => {
+      console.info("input changed", e)
+      setInput(inputModel?.getValue())
+    })
+    setInput(inputModel.getValue())
+  }, [inputEditor, inputModel])
+
+  const onMountOutput = useCallback(
+    (editor: editor.IStandaloneCodeEditor, monaco: Monaco) => {
+      const fileRootPath = "file:///"
+      monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
+        ...monaco.languages.typescript.typescriptDefaults.getCompilerOptions(),
+        moduleResolution: 100 as any, //monaco.languages.typescript.ModuleResolutionKind.NodeJs,
```

Comment:
Raised https://github.com/microsoft/monaco-editor/pull/4545 to expose these properly, though unfortunately even with `3` (Node16) I still see squiggly red lines unless I do the hack above to reorganize the files to the root of the module.